Makefile

```
@@ -349,6 +349,7 @@ deps: $(shell find $(BASE_DIR) -name "go.sum")
 	$(SILENT)test -z $(GOMOCK_REFLECT_DIRS) || { echo "Found leftover gomock directories. Please remove them and rerun make deps!"; echo $(GOMOCK_REFLECT_DIRS); exit 1; }
 	$(SILENT)go mod tidy
 ifdef CI
+	$(SILENT)GOTOOLCHAIN=local go mod tidy || { >&2 echo "gotoolchain does not match with installed Go version" ; exit 1 ; }
```

When this fails, we should ideally suggest engineer how to resolve. Something like this:

```suggestion
	$(SILENT)GOTOOLCHAIN=local go mod tidy || { >&2 echo "Go toolchain does not match with installed Go version. This is a compatibility check that prevents breaking downstream builds. If you really need to update the toolchain version, ask in #forum-acs-golang" ; exit 1 ; }
```